File: GDTMobApp/GDTAds/GDTHybridAd/HybridAdViewController.m
//
//  HybridAdViewController.m
//  GDTMobApp
//
//  Created by royqpwang on 2018/12/20.
//  Copyright © 2018 Tencent. All rights reserved.
//

#import "HybridAdViewController.h"
#import "GDTAppDelegate.h"
#import "GDTHybridAd.h"

@interface HybridAdViewController () <GDTHybridAdDelegate>
@property (weak, nonatomic) IBOutlet UITextField *urlTextField;
@property (nonatomic, strong) GDTHybridAd *hybridAd;

@end

@implementation HybridAdViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.urlTextField.text = @"https://developers.adnet.qq.com/open/hybridH5?placement_id=5060996645157430&app_id=1109524399&xflow_lp_data=open_id%3D12345656%26nicename%3Dtencent&xflow_pos_id=1111";
}

- (IBAction)clickOpenUrl:(id)sender {
    self.hybridAd = [[GDTHybridAd alloc] initWithType:GDTHybridAdOptionRewardVideo];
    self.hybridAd.delegate = self;
    self.hybridAd.navigationBarColor = [UIColor whiteColor];
    self.hybridAd.titleFont = [UIFont systemFontOfSize:16];
    [self.hybridAd loadWithUrl:self.urlTextField.text];
    [self.hybridAd showWithRootViewController:self];
}

- (void)gdt_hybridAdDidPresented:(GDTHybridAd *)hybridAd
{
    NSLog(@"%s",__FUNCTION__);
    NSLog(@"浏览器展示成功");
}

- (void)gdt_hybridAdDidClose:(GDTHybridAd *)hybridAd
{
    NSLog(@"%s",__FUNCTION__);
    NSLog(@"浏览器关闭");
    self.hybridAd = nil;
}

- (void)gdt_hybridAdLoadURLSuccess:(GDTHybridAd *)hybridAd
{
    NSLog(@"%s",__FUNCTION__);
    NSLog(@"URL 加载成功");
}

- (void)gdt_hybridAd:(GDTHybridAd *)hybridAd didFailWithError:(NSError *)error
{
    NSLog(@"%s",__FUNCTION__);
    if ([error code] == 3001) {
        NSLog(@"URL 加载失败");
    }
}


@end

File: GDTMobSample-Swift/GDTAdTypes/HybridAdViewController/HybridAdViewController.swift
//
//  HybridAdViewController.swift
//  GDTMobSample-Swift
//
//  Created by royqpwang on 2019/3/9.
//  Copyright © 2019 Tencent. All rights reserved.
//

import UIKit

class HybridAdViewController: UIViewController, GDTHybridAdDelegate {
    
    @IBOutlet weak var urlTestField: UITextField!
    override func viewDidLoad() {
        
    }
    @IBAction func clickLoad(_ sender: Any) {
        GDTSDKConfig.registerAppId(Constant.appID)
        let hybridAD = GDTHybridAd.init(type: GDTHybridAdOptions.rewardVideo)
        hybridAD.delegate = self
        hybridAD.load(withUrl: self.urlTestField.text!)
        hybridAD.show(withRootViewController: self)
    }
    
    func gdt_hybridAdDidPresented(_ hybridAd: GDTHybridAd) {
        print("浏览器展示成功")
    }
    
    func gdt_hybridAdDidClose(_ hybridAd: GDTHybridAd) {
        print("浏览器关闭")
    }
    
    func gdt_hybridAdLoadURLSuccess(_ hybridAd: GDTHybridAd) {
        print("URL 加载成功")
    }
    
    func gdt_hybridAd(_ hybridAd: GDTHybridAd, didFailWithError error: Error) {
        let code = (error as NSError).code
        if (code == 3001) {
            print("加载失败")
        }
    }
}
